systemtest/src/test/java/io/strimzi/systemtest/upgrade/OlmUpgradeST.java

```diff
@@ -0,0 +1,212 @@
+
+    private final String namespace = "olm-upgrade-namespace";
+    private final String producerName = "producer";
+    private final String consumerName = "consumer";
+    private final String topicUpgradeName = "topic-upgrade";
+    private final int messageUpgradeCount =  10_000;
+    private final Map<String, List<String>> mapOfKafkaVersionsWithSupportedClusterOperators = getMapKafkaVersionsWithSupportedClusterOperatorVersions();
+
+    @Test
+    void testUpgrade() {
+        Map<String, String> kafkaSnapshot = null;
+        boolean isUpgradeAble = true;
+
+        while (isUpgradeAble) {
+            // 1. Create subscription (+ operator group) with version latest - 1 (manual approval strategy) already done...!
+            // 2. Approve installation
+            //   a) get name of install-plan
+            //   b) approve installation
+            OlmResource.upgradeAbleClusterOperator(namespace, OlmInstallationStrategy.Manual, false);
+
+            String currentVersionOfCo = OlmResource.getClusterOperatorVersion();
+
+            LOGGER.info("====================================================================================");
+            LOGGER.info("============== Verification version of CO:" + currentVersionOfCo);
+            LOGGER.info("====================================================================================");
+
+            // wait until RU is finished (first run skipping)
+            if (kafkaSnapshot != null) {
+                StatefulSetUtils.waitTillSsHasRolled(KafkaResources.kafkaStatefulSetName(CLUSTER_NAME), 3, kafkaSnapshot);
+            }
+
+            // 3. perform verification of specific version
+            performUpgradeVerification();
+            kafkaSnapshot = StatefulSetUtils.ssSnapshot(KafkaResources.kafkaStatefulSetName(CLUSTER_NAME));
+
+            OlmResource.getClosedMapInstallPlan().put(OlmResource.getNonUsedInstallPlan(), Boolean.TRUE);
+            OlmResource.obtainInstallPlanName();
+            isUpgradeAble = OlmResource.isUpgradeable();
+        }
+    }
+
+    private void performUpgradeVerification() {
+        // fetch the tag from imageName: docker.io/strimzi/operator:'[latest|0.19.0|0.18.0]'
+        String containerImageTag = kubeClient().getDeployment(kubeClient().getDeploymentNameByPrefix(Constants.STRIMZI_DEPLOYMENT_NAME))
+            .getSpec()
+            .getTemplate()
+            .getMetadata()
+            .getAnnotations()
+            .get("containerImage").split(":")[1];
+
+        LOGGER.info("Image tag of strimzi operator is {}", containerImageTag);
+
+        // NOT (latest image or default substring(1)) for skipping 'v'0.19.0 on the start...
+        // '6.6.6' is the latest version of cluster operator
+        if (!containerImageTag.equals("6.6.6") && (!(containerImageTag.equals("latest") || containerImageTag.equals(Environment.OLM_OPERATOR_VERSION_DEFAULT.substring(1))))) {
```

Comment:
`6.6.6` should be passed to STs via som env var. In case the latest version in our image will change, we case easily forget to chage it here and it could take some time to debug